### DISCUSSION TITLE
Redis 和 I/O 多路复用 · /redis-io-multiplexing

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#270] [Announcements]
https://draveness.me/redis-io-multiplexing

**Comment by qci133** [comment]
这篇文章讲得深入浅出，看得出作者功底很深。顺便问一句，文章中的配图风格很好看，是怎么制作的呀？

**Comment by learningChild** [comment]
赞

**Comment by kenwoodjw** [comment]
写得很好

**Comment by rapharino** [comment]
使用的画图工具叫什么?

**Comment by wujunze** [comment]
> 使用的画图工具叫什么?
文章底部有说明
图片使用 Sketch 进行绘制

**Comment by ssy341** [comment]
配图的文字能正楷一点就好了，看起来太费劲了

**Comment by oneweek20169902** [comment]
请问下楼主建站是用的什么工具

**Comment by draveness** [comment]
jekyll

**Comment by 0xiaobao0** [comment]
支持~

**Comment by linuxjack** [comment]
https://juejin.im/post/5ccaef8cf265da038932a357#heading-8 这篇文章没注明出处哦，是你授权的吗？

**Comment by tw7613781** [comment]
目前在学习系统编程，苦于一直没有找到学习材料，感到迷茫没有方向。谢谢深入浅出的文章～

**Comment by GingJan** [comment]
可否讲解下 select、epoll、kqueue这三多路IO函数的底层模型和实现机制呢？特别是epoll、kqueue这些是通过什么方式实现的O(1)？谢谢

**Comment by 123456ZYF88** [comment]
在高并发的时候 select 不是应该比较慢吗  时间复杂度 O（n）
